src/api.jl

```diff
@@ -302,19 +302,19 @@ end
 function (p::MapParameterValue)(k, observer=nothing; kwargs...)
     i = _search_equal(p.value.indexes, k)
     i === nothing && return p(; kwargs...)
-    pvs = p.value.values[i]
+    pvs = parameter_value(p.value.values[i])
```

Comment:
Does this mean we have to build a MapParameterValue everytime we call a MapParameterValue? We typically call these functions a lot from SpineOpt, like once per time-slice scenario combination and several times. Building the MapParameterValue is somewhat expensive so I'd prefer not to do it repeatedaly if possible?

Reply:
@manuelma 

Yeah, I know this is not ideal, and am aware it might cause some performance issues. Unfortunately, it was the cleanest way I could think of for ensuring that `Map` arithmetic works easily when the `Map`s are accessed through `MapParameterValue` calls.

The root of the problem is, that the old way of doing things recursively converted all the contents of the `Map` into their `parameter_value()` equivalents as soon as a `MapParameterValue` was formed. This meant that if you built a `Map`, made it into a `MapParameterValue`, and called it, it didn't return the original `Map`, but one where its contents are `AbstractParameterValue`s instead. Essentially, `Map` arithmetic would not work for `Map`s accessed via `MapParameterValue` calls.

Essentially, we currently have a bit of a mess whether `Map`s contain data types like `Real, TimeSeries, TimeValue, Map`, or their `AbstractParameterValue` conterparts, as they can do both depending on how the `Map` was built and accessed.

Reply:
I suppose an alternative would be to keep the recursive `parameter_value()` when initially forming the `MapParameterValue`s when reading the database, but then recursively `getfield(x, :value)` to get rid of the potential nested `AbstractParameterValues` when returning the final result of the parameter call. Either that, or implement identical arithmetic for the `AbstractParameterValue` counterparts of the data types.

Reply:
Ok, I think I managed to fix this while avoiding excessive `parameter_value` calls. Needed to implement a new `_recursive_inner_value` function for returning the original `Map` after a `MapParamereterValue` call. The crux of the issue is implemented as a new unit test:
```
parameter_value(map)() == map
```
which was previously not true.

Reply:
yes, that's what we need I think - good one @Tasqu